CodePushPackage.m

```diff
@@ -204,15 +204,22 @@ + (void)downloadPackage:(NSDictionary *)updatePackage
            failCallback:(void (^)(NSError *err))failCallback
 {
     NSString *newPackageFolderPath = [self getPackageFolderPath:updatePackage[@"packageHash"]];
-    NSError *error = nil;
+    NSError *error;
     
-    if (![[NSFileManager defaultManager] fileExistsAtPath:newPackageFolderPath]) {
-        [[NSFileManager defaultManager] createDirectoryAtPath:newPackageFolderPath
+    if (![[NSFileManager defaultManager] fileExistsAtPath:[self getCodePushPath]]) {
```

Comment:
Previously, we depended on the fact that we create the newPackageFolderPath if it doesnt exist to create the intermediate directories. Now, we only delete it if it exists, so we need this addition to create the intermediate directories, otherwise, the download handler (NSURLConnection) will fail